Disable native maven based tests

timja opened the pull request:
<!-- Comment: 
A great PR typically begins with the line below.
Replace XXXXX with the numeric part of the issue's id you created on JIRA.
Please note that if you want your changes backported into LTS, you will need to create a JIRA ticket for it. Read https://jenkins.io/download/lts/#backporting-process for more.
-->

<!-- Comment: 
If the issue is not fully described in the ticket, add more information here (justification, pull request links, etc.).

 * We do not require JIRA issues for minor improvements.
 * Bugfixes should have a JIRA issue (backporting process).
 * Major new features should have a JIRA issue reference.
-->

### Proposed changelog entries

* Internal: Change tests to retrieve maven over https due to upstream requiring https only
* ...

<!-- Comment: 
The changelogs will be integrated by the core maintainers after the merge.  See the changelog examples here: https://jenkins.io/changelog/ -->

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [na] JIRA issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

<!-- For new API and extension points: Link to the reference implementation in open-source (or example in Javadoc) -->

### Desired reviewers

@jenkinsci/core-pr-reviewers 

<!-- Comment:
If you need an accelerated review process by the community (e.g., for critical bugs), mention @jenkinsci/code-reviewers
-->

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

timja committed:
Fix maven tests - https only

oleg-nenashev reviewed (approved):
Thanks for the quick fix! I will merge it once the build passes

timja commented:
I think more changes are needed unfortunately 😢 
https://stackoverflow.com/a/59764670/4951015

Only recent maven uses https by default

timja committed:
More test fixes

timja commented:
The specific test that is failing passes locally, might be to do with the maven version running on CI

timja commented:
Managed to reproduce locally,
I had maven-clean-plugin:2.2 cached in my local repository, and then after removing it, it was using my jenkins-ci mirror I have configured, after disabling that I can reproduce it

timja committed:
Copy settings.xml to home directory

Due to the maven central change to require only https, this ensures that
tests pickup the settings as well and not just the main process

timja commented:
Might need to get someone else to re-submit this, cc @jenkinsci/core 
```
19:42:33  Obtained Jenkinsfile from d02c7563588aa534629e05b873946680953dbfdc
[2020-01-17T19:42:34.292Z] ‘Jenkinsfile’ has been modified in an untrusted revision
```

oleg-nenashev commented:
on it

oleg-nenashev commented:
If #4437 does not play out, will just mark tests as ignored for now

timja committed:
Revert "Copy settings.xml to home directory"

This reverts commit 37ccab7c55dbd4ab8fe9cd9ad3265dc5fc2f5bc9.

timja committed:
Ignore tests that fail on CI due to https required on maven central

MarkEWaite commented on Jenkinsfile:
```diff
@@ -119,7 +114,7 @@ void withMavenEnv(List envVars = [], def buildType, def javaVersion, def body) {
         }
     }
-
+    
```
Could we avoid the whitespace change in this line?

timja replied:
it's been reverted

MarkEWaite reviewed (approved):
Minor comment about restoring a line with trailing whitespace. Does not block the merge

oleg-nenashev reviewed (approved):
Would be nice to have a follow-up task in Jira. Will merge once the build passes

oleg-nenashev commented:
One test is still failing. I might find some time to fix/ignore it tonight, but no committment.

timja committed:
Comment out HudsonTestCase based test

timja commented:
> One test is still failing. I might find some time to fix/ignore it tonight, but no committment.

https://github.com/jenkinsci/jenkins/pull/4435/commits/6a34d5fb9d91acdee9e2f3bc92c994dfd8ff4246

`@Ignore doesn't work with junit.framework.TestCase` based tests

timja committed:
Rerun CI

timja commented:
Build is green after the last kick

Follow-up jira issue:
https://issues.jenkins-ci.org/browse/JENKINS-60813

oleg-nenashev commented:
Thanks @timja!
